indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -235,6 +238,25 @@ public void onFailure(Throwable throwable)
                   removeWorker(worker);
                   break;
                 case INITIALIZED:
+                  // Schedule cleanup for task status of the workers that might have disconnected while overlord was not running
+                  List<String> workers;
+                  try {
+                    workers = cf.getChildren().forPath(indexerZkConfig.getStatusPath());
+                  } catch (KeeperException.NoNodeException e) {
+                    // statusPath doesn't exist yet; can occur if no middleManagers have started.
+                    workers = ImmutableList.of();
+                  }
+                  for (String workerId : workers) {
+                    if (!zkWorkers.containsKey(workerId)
+                        && cf.checkExists().forPath(JOINER.join(indexerZkConfig.getAnnouncementsPath(), workerId)) == null) {
+                      scheduleTasksCleanupForWorker(
+                          workerId,
+                          cf.getChildren()
+                            .forPath(JOINER.join(indexerZkConfig.getStatusPath(), workerId))
```

Comment:
will getChildren return null if the path does not exist? if so you'll end up with NPE in the method call